openml-utils/src/test/java/com/feedzai/util/provider/AbstractProviderModelBaseTest.java

```diff
@@ -136,34 +137,46 @@ public void createTwoModelsInOneThreadTest() throws ModelLoadingException, Model
     }
 
     /**
-     * Evaluates one model created (loaded/trained) in the main thread in two threads concurrently.
+     * Evaluates one model created (loaded/trained) in the main thread in multiple threads concurrently.
+     * The model will evaluate two types of instances, that will be injected at the same frequency (half of
+     * {@link #maxNumberOfThreads}).
      *
      * @throws ModelLoadingException If anything goes wrong during loading.
      * @throws ModelTrainingException If anything goes wrong during training.
      */
     @Test
-    public void createOneModelAndEvaluateInTwoThreadsTest() throws ModelLoadingException, ModelTrainingException {
+    public void createOneModelAndEvaluateInMultipleThreadsTest() throws ModelLoadingException, ModelTrainingException {
         final ExecutorService executor = Executors.newFixedThreadPool(this.maxNumberOfThreads);
         final M firstModel = getFirstModel();
 
-        final List<Future<Integer>> evaluationsList = classifyInstancesInParallel(firstModel, executor);
+        final List<Future<double[]>> evaluationsList = classDistTwoInstancesInParallel(firstModel, executor);
 
         // Wait for all classifications to finish
-        final List<Integer> classification = getUnorderedClassificationResults(evaluationsList);
+        final List<double []> classification = getUnorderedClassificationResults(evaluationsList);
 
-        assertThat(classification)
-                .as("There is an error with the prediction value")
-                .isSubsetOf(getClassifyValuesOfFirstModel().toArray(new Integer[0]));
+        final List<Double> collect = classification.stream()
```

Comment:
odd variable name "collect"

Reply:
Also "classification" is not reflecting properly the contents of the variable anymore; it's actually the classDistributionResults